src/envoy/http/mixer/check_data.cc

```diff
@@ -38,9 +38,12 @@ const std::set<std::string> RequestHeaderExclusives = {
 }  // namespace
 
 CheckData::CheckData(const HeaderMap& headers,
-                     const envoy::api::v2::core::Metadata& metadata,
-                     const Network::Connection* connection)
-    : headers_(headers), metadata_(metadata), connection_(connection) {
+                     const envoy::api::v2::core::Metadata& dynamic_metadata,
+                     const Network::Connection* connection,
+                     const envoy::api::v2::core::Metadata& route_metadata,
+                     const envoy::api::v2::core::Metadata& cluster_metadata)
+    : headers_(headers), dynamic_metadata_(dynamic_metadata), connection_(connection),
+    route_metadata_(route_metadata), cluster_metadata_(cluster_metadata) {
```

Comment:
Cluster metadata is interesting for Check. I think if a re-route happens in a subsequent filter (or using route directive), then this selection of cluster is likely to be wrong.

Reply:
Moved to Report.